### PR TITLE
#3751: Esm interop flag

### PR DESCRIPTION
**Pull request by D10221**
Enable esm interop behind a flag
Enable experimental, optional use
Minimum changes
Tests: migration/seeds/configuration
fix:  eslint error

### CODE REVIEW

File: test/jake/jakelib/migrate-test.js [hunk 1]
```diff
@@ -734,6 +734,20 @@ test('migrate:list prints migrations both completed and pending', async (temp) =
...
+test('migrate runs "esm" modules', (temp) => {
+  const db = knexfile.connection.filename;
+  if (fs.existsSync(db)) {
+    fs.unlinkSync(db);
+  }
+
+  return assertExec(
+    `node ${KNEX} --esm migrate:latest --knexfile=test/jake-util/knexfile-esm/knexfile.js --knexpath=../knex.js`
+  ).then(({ stdout }) => {
+    assert.include(stdout, 'Batch 1 run: 1 migrations');
```
Comment (kibertoad) [inline test/jake/jakelib/migrate-test.js]: `cli-testlab 1.10.0` supports multiple assertions, so if you bump dependency, you can do

```
return assertExec(
    `node ${KNEX} --esm migrate:latest --knexfile=test/jake-util/knexfile-esm/knexfile.js --knexpath=../knex.js`, {
      expectedOutput: ['Batch 1 run: 1 migrations', 'one.js']
    }
  )
```

File: test/jake/jakelib/seed-test.js [hunk 2]
```diff
@@ -55,6 +55,16 @@ test(taskList, 'Handles seeding errors correctly', (temp) => {
   });
 });
 
+test(taskList, 'seed:run runs "esm" files', () => {
+  return assertExec(
+    `node ${KNEX} --esm seed:run --knexfile=test/jake-util/knexfile-esm/knexfile.js`
+  ).then(({ stdout }) => {
+    assert.include(stdout, 'Ran 1 seed files');
+    assert.notInclude(stdout, 'first.js');
```
Comment (kibertoad) [inline test/jake/jakelib/seed-test.js]: you can use `notExpectedOutput` and `expectedOutput` params for `assertExec` to avoid `then` chaining

**Comment by kibertoad** [comment]
Thanks! There are tiny little things left, and then this can be merged :)

**Comment by kibertoad** [comment]
Could you also please submit a PR with documentation for this on https://github.com/knex/documentation?

**Comment by D10221** [comment]
Thanks, 
Would  you like me to implement the remarks, verbatim? 
Or its an opportunity ? :)
I'll check the docs

**Comment by kibertoad** [comment]
@D10221 oh, you are right, this particular file is not using cli-testlab yet, so please ignore that comment, I'll refactor myself later.

**Comment by D10221** [comment]
Thanks.
Opened a doc's [PR](https://github.com/knex/documentation/pull/251)
Let me know if its aligned to 'expectations' :)

**Comment by kibertoad** [comment]
Released in 0.20.8

**Pull request merged**